src/Raven.Server/Config/Categories/ConfigurationCategory.cs

```diff
@@ -52,34 +52,42 @@ string GetValue(IConfiguration cfg, string name)
                     : null;
             }
 
-            string GetConfiguration(IConfiguration cfg, string name)
+            string GetConfigurationValue(IConfiguration cfg, string keyName, out bool keyExistsInConfiguration)
             {
-                if (cfg == null || name == null)
+                keyExistsInConfiguration = false;
+
+                if (cfg == null || keyName == null)
                     return null;
 
-                var val = GetValue(cfg, name);
+                // This check is needed because cfg.GetSection(keyName) returns null even if key does Not exist in configuration!
+                keyExistsInConfiguration = cfg.AsEnumerable().Any(x => string.Equals(x.Key, keyName));
```

Comment:
string.Equals(x.Key, keyName, StringComparer.OrdinalIgnoreCase)

Reply:
done - for both places where used - using _StringComparison.OrdinalIgnoreCase_